test/e2e/util_test.go

```diff
@@ -509,6 +509,18 @@ func buildCatalogSourceCleanupFunc(crc versioned.Interface, namespace string, ca
 		ctx.Ctx().Logf("Deleting catalog source %s...", catalogSource.GetName())
 		err := crc.OperatorsV1alpha1().CatalogSources(namespace).Delete(context.Background(), catalogSource.GetName(), metav1.DeleteOptions{})
 		Expect(err).ToNot(HaveOccurred())
+
+		err = wait.Poll(pollInterval, pollDuration, func() (bool, error) {
+			fetched, err := newKubeClient().KubernetesInterface().CoreV1().Pods(catalogSource.GetNamespace()).List(context.Background(), metav1.ListOptions{LabelSelector: "olm.catalogSource=" + catalogSource.GetName()})
+			if err != nil {
+				ctx.Ctx().Logf("Error: getting catalog source pod: %s", err)
+				return false, err
+			}
+			if len(fetched.Items) == 0 {
+				return true, nil
+			}
+			return false, nil
+		})
```

Comment:
Just coming back to this:

- It doesn't look like we're handling the wait.Poll error anywhere.
- We should try and aim for a better logging as well

WDYT about the following:

```go
Eventually(func() (bool, error) {
	fetched, err := newKubeClient().KubernetesInterface().CoreV1().Pods(catalogSource.GetNamespace()).List(context.Background(), metav1.ListOptions{LabelSelector: "olm.catalogSource=" + catalogSource.GetName()})
	if err != nil {
		return false, err
	}
	if len(fetched.Items) == 0 {
		return true, nil
	}
	ctx.Ctx().Logf("waiting for the catalog source %s pod to be deleted...", fetched.GetName())
	return false, nil
}).Should(BeTrue())
```

Reply:
Thanks!  I'll take your changes.